specs/angular-readonly-form-styling/README.md

```diff
@@ -0,0 +1,152 @@
+![disabled form](disabled-form.png)
+
+Improved UI with read-only controls:
+![readonly form](readonly-form.png)
+
+Note that in the improved UI screenshot, there are likely still additional improvements that should be made. These improvements include, but are not limited to, updating the `nimble-switch` and `nimble-checkbox` instances to no longer be disabled. However, a decision has not yet been made regarding how these components should be represented when read-only.
+
+## Links To Relevant Work Items and Reference Material
+
+[SLE control pattern docs](https://stratus-storybook.ni.dev/?path=/docs/patterns-control-text--docs)
+
+## Implementation / Design
+
+To enable this new design, the following changes need to be made within the nimble repo, each of which is discussed in more detail below:
+
+- Introduce `readonly` configuration on the `nimble-select` and `nimble-combobox`
+- Introduce `frameless` appearance to the `nimble-select`, `nimble-combobox`, and `nimble-number-field`
+- Introduce `full-bleed` configuration on the `nimble-select`, `nimble-combobox`, and `nimble-number-field`
+- Introduce a `readOnlyWhenDisabled` configuration on the control value accessors in Angular for the `nimble-select`, `nimble-combobox`, `nimble-text-area`, `nimble-text-field`, and `nimble-number-field`
+
+### Read-only select and combobox
+
+The `nimble-select` and `nimble-combobox` will be updated to have a new boolean attribute named `readonly`, which will be associated with the `readOnly` boolean property.
+
+When these components are read-only, they will have the following characteristics:
+
+- Tab focusable
+- Cannot change their value using the mouse or keyboard
+- Cannot open their popup
+- Drop-down arrow (select) and drop-down button (combobox) will be hidden
+- ARIA updates when read-only:
+    - `aria-haspopup` set to `null`
+    - `aria-readonly` set to `"true"`
+    - Note: While the native select does not support `readonly`, the `combobox` role does support `aria-readonly`, so that can be used in combination with the existing role of both the `nimble-select` and `nimble-combobox`. With these two changes locally, storybook does not report any new accessibility violations.
+
+### Frameless select, combobox, and number field
+
+The visual design for read-only controls in forms is for them to be rendered without any borders or backgrounds. This matches the `frameless` appearance nimble has today for the `nimble-text-field`. Therefore, the select, combobox, and number field will all be updated to also support a `frameless` appearance.
+
+### Full-bleed select, combobox, and number field
+
+The visual design for read-only controls in forms is for them to be rendered without any left padding on the value (i.e. the value is aligned with the label). This matches the `full-bleed` configuration nimble has today for the `nimble-text-field`. Therefore, the select, combobox, and number field will all be updated to also support being `full-bleed`.
```

Comment:
Is it worth noting that, like the `nimble-text-field`, the `full-bleed` configuration only applies to the `frameless` appearance?

Reply:
Yes, that's a great point. My plan is to have `full-bleed` + `frameless` to work together the same way as it does on the `nimble-text-field` today. I should explicitly call that out here.